### ISSUE TITLE
Nvim opens a [No Name] buffer when started on a directory

### ISSUE THREAD

**Issue by mig-hub** [neovim/neovim#17841]
### Neovim version (nvim -v)

NVIM v0.6.1

### Vim (not Nvim) behaves the same?

no, Works fine on Vim 8.2

### Operating system/version

macOS 12.1 Monterey

### Terminal name/version

Terminal.app 2.12

### $TERM environment variable

xterm-256color

### Installation

Homebrew

### How to reproduce the issue

```
nvim -u NORC .
```

### Expected behavior

This should start `nvim` with a single netrw buffer.
The status line should indicate the path followed by "[netrw]".

### Actual behavior

It opens 2 buffers, the netrw one and a "[No Name]" one.
The status line mixes both names and says "[No Name] [netrw]"

Essentially it seems like you see the right netrw buffer and can interact with it, except nvim kind of believe it is on the "[No Name]" buffer.

The main issue for me is that it also places the "[No Name]" buffer as the current buffer. So if I open a file in the directory listing and then press `CTRL-^`, I expect to be back on the netrw buffer. But I end up on the real "[No Name]" buffer that is empty. I say "real" because when landing, it thinks I am on "[No Name]" but what I see and interact with is really the netrw buffer.

**Comment by mig-hub** [comment]
@zeertzjq indeed if I `set nohidden` in my vimrc, it seems to work fine. Thank you. 

Is it still a bug ? or is the behaviour I've described the intended behaviour with a side effect that I don't understand?

**Comment by justinmk** [comment]
works for me on 9e4286294f9c

if it's still an issue please comment. Would likely be a netrw bug

**Comment by strager** [comment]
I hit this issue too. Adding `set nohidden` to my `init.vim` fixed the issue and made Neovim behave like Vim's default.

**Comment by mukul98s** [comment]
is there any solution to this issue? It is causing a lot of trouble for the for the past 2 weeks.

**Comment by strager** [comment]
> is there any solution to this issue?

Yes: https://github.com/neovim/neovim/issues/17841#issuecomment-1152837090

**Comment by mukul98s** [comment]
@strager I have tried this method. This does not work for me.

**Comment by georgevpanayotov** [comment]
I've got a workaround. I haven't tried this in neovim just regular vim so ymmv.

```
au FileType netrw call UpdateNetrwBuffer()

" Workaround to the netrw buffers being listed as '[No Name]'
function UpdateNetrwBuffer()
    " This issue happens only when 'hidden' is true. It seems that netrw will create a new buffer to
    " load its browser. However, when opening a directory vim will create an empty buffer with that
    " directory's name. If 'nohidden' then this buffer is deleted as soon as the netrw buffer is
    " created, this allows netrw to rename its buffer to the directory's name. If 'hidden' then the
    " pre-existing empty buffer hangs around and interferes with the attempt to rename the netrw
    " buffer. This fix just deletes the empty buffer and renames the netrw buffer.
    exec bufnr(b:netrw_curdir) . 'bd'
    exec 'file ' . b:netrw_curdir
endfunction
```

Edit: you'll want to check:
```
    if bufnr(b:netrw_curdir) != bufnr('%')
```
to avoid deleting valid buffers.

**Comment by Sherlockouo** [comment]
> I've got a workaround. I haven't tried this in neovim just regular vim so ymmv.
> 
> ```
> au FileType netrw call UpdateNetrwBuffer()
> 
> " Workaround to the netrw buffers being listed as '[No Name]'
> function UpdateNetrwBuffer()
>     " This issue happens only when 'hidden' is true. It seems that netrw will create a new buffer to
>     " load its browser. However, when opening a directory vim will create an empty buffer with that
>     " directory's name. If 'nohidden' then this buffer is deleted as soon as the netrw buffer is
>     " created, this allows netrw to rename its buffer to the directory's name. If 'hidden' then the
>     " pre-existing empty buffer hangs around and interferes with the attempt to rename the netrw
>     " buffer. This fix just deletes the empty buffer and renames the netrw buffer.
>     exec bufnr(b:netrw_curdir) . 'bd'
>     exec 'file ' . b:netrw_curdir
> endfunction
> ```
> 
> Edit: you'll want to check:
> 
> ```
>     if bufnr(b:netrw_curdir) != bufnr('%')
> ```
> 
> to avoid deleting valid buffers.

Thanks for your solution! I wrote a neovim version, this works for me.

```lua
opts = function(_, opts)
  local edgy_idx = LazyVim.plugin.extra_idx("ui.edgy")
  local symbols_idx = LazyVim.plugin.extra_idx("editor.outline")

  if edgy_idx and edgy_idx > symbols_idx then
    LazyVim.warn(
      "The `edgy.nvim` extra must be **imported** before the `outline.nvim` extra to work properly.",
      { title = "LazyVim" }
    )
  end

  opts.right = opts.right or {}
  table.insert(opts.right, {
    title = "Outline",
    ft = "Outline",
    pinned = true,
    open = "Outline",
  })
end
```
